ds3_java_cli/src/main/java/com/spectralogic/ds3cli/command/PutBulk.java

```diff
@@ -141,9 +159,44 @@ public PutBulkResult call() throws Exception {
 //            job.withRequestModifier(new ComputedChecksumModifier());
         }
         LOG.info("Created bulk put job " + job.getJobId().toString() + ", starting transfer");
-        job.transfer(new PrefixedFileObjectPutter(this.inputDirectory, prefix));
 
-        return new PutBulkResult("SUCCESS: Wrote all the files in " + this.inputDirectory.toString() + " to bucket " + this.bucketName);
+        if (pipe) {
+            job.transfer(new Ds3ClientHelpers.ObjectChannelBuilder() {
```

Comment:
To make this cleaner, this should probably be either it's own standalone class, or an inner class in the file, but not an anonymous inner class.